src/server/registry/catalog.go

```diff
@@ -129,6 +140,34 @@ func (r *repositoryHandler) sendResponse(w http.ResponseWriter, req *http.Reques
 	}
 }
 
+// empty repo and all of artifacts are untagged should be filtered out.
+func (r *repositoryHandler) validateRepo(ctx context.Context, repositoryID int64) (bool, error) {
+	arts, err := r.artCtl.List(ctx, &q.Query{
+		Keywords: map[string]interface{}{
+			"RepositoryID": repositoryID,
+		},
+	}, &artifact.Option{
+		WithTag: true,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	// empty repo
+	if len(arts) == 0 {
+		return false, nil
+	}
+
+	for _, art := range arts {
+		if len(art.Tags) != 0 {
+			return true, nil
+		}
+	}
+
+	// if all of artifact are untagged, filter out
+	return false, nil
```

Comment:
How about using `Count` method? It's more lightweight than `List`

Reply:
but, it needs two DB calls. The `list` cannot be avoid.

Reply:
per offline discussion, use `list` to query artifact.